kratos/processes/simple_mortar_mapper_process.cpp

```diff
@@ -29,8 +29,8 @@ SimpleMortarMapperProcess<TDim, TNumNodes, TVarType, THistOrigin, THistDestinati
     TVarType& ThisVariable,
     Parameters ThisParameters,
     LinearSolverType::Pointer pThisLinearSolver
-    ): mrOriginModelPart(*new ModelPart()),
-       mrDestinationModelPart(*new ModelPart()),
+    ): mOriginModelPart(ModelPart("OriginModelPart")),
```

Comment:
why not Kratos::make_shared?

Reply:
guys, the point is that i would like to have as a mechanism that when a model part is created, it is automatically added to the Model. to achieve this it is mandatory that we avoid "strange" construction mechanisms

Reply:
@loumalouomega  this also means that you should not allocate a new modelpart there. Can't you just deprecate this constructor and oblige users to pass it constructed from outside?

Reply:
Well, that can be done, but will require some changes